test-data/unit/check-del-variable-type.test

```diff
@@ -0,0 +1,42 @@
+[case testDelVariableResetsType]
```

Comment:
There are only few tests needed for this feature, let's find a relevant existing file and move the tests there instead of adding a new file. `check-statements.test` looks like a reasonable place, somewhere in "Del statement" block.